java/apps/authentication-portal/src/main/webapp/add-security-questions.jsp

```diff
@@ -97,16 +97,28 @@
 </head>
 <body class="login-portal layout authentication-portal-layout">
     <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
-        <layout:component componentName="ProductHeader" >
+        <layout:component componentName="ProductHeader">
             <%-- product-title --%>
             <%
-                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+            File productTitleFile = null;
+            String productTitleFilePath;
+            
+            if (StringUtils.isNotBlank(customLayoutFileRelativeBasePath)) {
+                productTitleFile = new File(getServletContext().getRealPath(customLayoutFileRelativeBasePath + "/product-footer.jsp"));
+            }
+
+            if (productTitleFile != null && productTitleFile.exists()) {
+                productTitleFilePath = customLayoutFileRelativeBasePath + "/product-title.jsp";
+            } else {
+                productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
                 if (productTitleFile.exists()) {
+                    productTitleFilePath = "extensions/product-title.jsp";
+                } else {
+                    productTitleFilePath = "includes/product-title.jsp";
+                }
+            }
```

Comment:
Any possibility we can create a reusable method for this, instead of duplicating the logic in all files?

Reply:
Since this change is to define another extension point, we cannot move this logic to extensions/product-title.jsp